backend/internal/service/server.go

```diff
@@ -116,7 +116,7 @@ func setupRoutes(app *fiber.App, config config.Config) {
 		h := spotify_handler.NewHandler(repository.Media)
 		m := spotify_middleware.NewMiddleware(config.Spotify, repository.UserAuth, sessionStore)
 
-		r.Route("/:userID", func(authRoute fiber.Router) {
+		r.Route("/user/:userID", func(authRoute fiber.Router) {
```

Comment:
ok sorry to bring this PR back but there was a merge conflict w the routes after you split up the methods so i pulled locally to resolve it and then run it to make sure I didn't break anything

and when I tried to hit /spotify/import/recommendations it was trying to parse "import" as a userId and failing

so anyway I think adding `/user` fixes that issue, but while running I saw an error abt foreign keys get printed:

my request:
http://127.0.0.1:8080/spotify/import/recommendations
body:
`{
    "artists": ["00FQb4jTyendYWaN8pK0wa"]
}`

`2024/11/03 22:46:05 WARN error importing recommendations errors="ERROR: duplicate key value violates unique constraint \"album_spotify_id_key\" (SQLSTATE 23505)\nERROR: duplicate key value violates unique constraint \"album_spotify_id_key\" (SQLSTATE 23505)\nERROR: duplicate key value violates unique constraint \"album_spotify_id_key\" (SQLSTATE 23505)\nERROR: duplicate key value violates unique constraint \"album_spotify_id_key\" (SQLSTATE 23505)\nERROR: insert or update on table \"album_artist\" violates foreign key constraint \"album_artist_artist_id_fkey\" (SQLSTATE 23503)\nERROR: duplicate key value violates unique constraint \"album_spotify_id_key\" (SQLSTATE 23505)\nERROR: insert or update on table \"track_artist\" violates foreign key constraint \"track_artist_artist_id_fkey\" (SQLSTATE 23503)\nERROR: insert or update on table \"track_artist\" violates foreign key constraint \"track_artist_artist_id_fkey\" (SQLSTATE 23503)\nERROR: insert or update on table \"album_artist\" violates foreign key constraint \"album_artist_artist_id_fkey\" (SQLSTATE 23503)\nERROR: duplicate key value violates unique constraint \"album_spotify_id_key\" (SQLSTATE 23505)\nERROR: insert or update on table \"album_artist\" violates foreign key constraint \"album_artist_artist_id_fkey\" (SQLSTATE 23503)\nERROR: insert or update on table \"track_artist\" violates foreign key constraint \"track_artist_artist_id_fkey\" (SQLSTATE 23503)\nERROR: duplicate key value violates unique constraint \"album_spotify_id_key\" (SQLSTATE 23505)"
[22:46:01] 127.0.0.1:61553 94936 67920a43-c2e6-4f8a-a5c2-a187cf6a3f1b 200 -  3.838368209s POST /spotify/import/recommendations`

can you look into this when you have a chance? not super urgent

Reply:
changed code so it expects unique violation errors and doesnt do anything. for other errors, it will log them. 

also, i think its okay for errors to happen -- the way the pipeline code is set up, a record should never be inserted with a missing foreign key reference

also, suppose i insert an album and get the id back, then i try to insert the corresponding track with the album id i got back and this operation fails. killing the job on the error here doesn't really help since the data was already inserted anyway. so i think its fine that we get errors. we just have to make sure data is inserted in correct order and if an operation fails mid way it doesnt mean the inserted data is invalid